tests/test_allpairs.py

```diff
@@ -82,24 +84,102 @@ def test_normal(self):
+            ["Brand X", "XP", "Internal", "Contr.", 15],
+            ["Brand X", "NT", "Internal", "Hourly", 15],
+            ["Brand Y", "NT", "Internal", "Contr.", 6],
+            ["Brand Y", "2000", "Internal", "Contr.", 10],
+            ["Brand Y", "NT", "Internal", "Salaried", 6],
+            ["Brand Y", "2000", "Internal", "Salaried", 30],
+            ["Brand Y", "NT", "Internal", "Contr.", 15],
+            ["Brand Y", "2000", "Internal", "Part-Time", 60],
+            ["Brand Y", "NT", "Internal", "Contr.", 30],
+            ["Brand Y", "2000", "Internal", "Salaried", 6],
             ["Brand Y", "NT", "Modem", "Salaried", 60],
-            ["Brand Y", "NT", "Modem", "Hourly", 15],
-            ["Brand Y", "NT", "Modem", "Hourly", 30],
-            ["Brand Y", "NT", "Modem", "Hourly", 60],
+            ["Brand Y", "2000", "Modem", "Hourly", 10],
+            ["Brand Y", "NT", "Internal", "Hourly", 30],
+            ["Brand Y", "2000", "Internal", "Contr.", 6],
+            ["Brand Y", "NT", "Modem", "Contr.", 10],
+            ["Brand Y", "2000", "Modem", "Part-Time", 15],
+            ["Brand Y", "NT", "Internal", "Part-Time", 6],
+            ["Brand Y", "2000", "Internal", "Hourly", 6],
+            ["Brand Y", "NT", "Modem", "Salaried", 30],
+            ["Brand Y", "2000", "Modem", "Salaried", 15],
+            ["Brand Y", "NT", "Internal", "Hourly", 60],
+            ["Brand Y", "2000", "Internal", "Part-Time", 30],
+            ["Brand Y", "NT", "Modem", "Part-Time", 60],
+            ["Brand Y", "2000", "Internal", "Hourly", 30],
+            ["Brand Y", "2000", "Internal", "Salaried", 10],
+            ["Brand Y", "2000", "Internal", "Contr.", 60],
+            ["Brand Y", "NT", "Modem", "Salaried", 15],
             ["Brand Y", "NT", "Modem", "Hourly", 10],
+            ["Brand Y", "NT", "Modem", "Part-Time", 10],
+            ["Brand Y", "2000", "Modem", "Contr.", 15],
+            ["Brand Y", "98", "Internal", "Contr.", 6],
+            ["Brand X", "NT", "Modem", "Contr.", 6],
+            ["Brand Y", "98", "Internal", "Salaried", 10],
+            ["Brand X", "NT", "Internal", "Part-Time", 60],
+            ["Brand Y", "98", "Internal", "Hourly", 30],
+            ["Brand X", "NT", "Internal", "Salaried", 30],
+            ["Brand Y", "98", "Internal", "Hourly", 15],
+            ["Brand X", "NT", "Internal", "Part-Time", 10],
+            ["Brand Y", "XP", "Internal", "Contr.", 60],
+            ["Brand X", "2000", "Internal", "Contr.", 6],
+            ["Brand Y", "XP", "Modem", "Part-Time", 6],
+            ["Brand Y", "XP", "Internal", "Hourly", 15],
+            ["Brand X", "2000", "Modem", "Salaried", 60],
+            ["Brand Y", "XP", "Internal", "Salaried", 10],
+            ["Brand Y", "XP", "Internal", "Hourly", 30],
+            ["Brand X", "2000", "Internal", "Part-Time", 15],
+            ["Brand X", "2000", "Internal", "Hourly", 30],
+            ["Brand X", "2000", "Internal", "Contr.", 10],
         ]
```

Comment:
Major increase in number of combinations when n > 2.

Currently, this test have the following input and expected result:
```python
parameters = [
	["Brand X", "Brand Y"],
	["98", "NT", "2000", "XP"],
	["Internal", "Modem"],
	["Salaried", "Hourly", "Part-Time", "Contr."],
	[6, 10, 15, 30, 60],
]

assert list(AllPairs(parameters, n=3)) == [
	["Brand X", "98", "Internal", "Salaried", 6],
	["Brand Y", "NT", "Modem", "Hourly", 6],
	["Brand Y", "2000", "Modem", "Part-Time", 10],
	["Brand X", "XP", "Internal", "Contr.", 10],
	["Brand X", "XP", "Modem", "Part-Time", 6],
	["Brand Y", "2000", "Internal", "Hourly", 15],
	["Brand Y", "NT", "Internal", "Salaried", 10],
	["Brand X", "98", "Modem", "Contr.", 15],
	["Brand X", "98", "Modem", "Hourly", 10],
	["Brand Y", "NT", "Modem", "Contr.", 30],
	["Brand X", "XP", "Internal", "Hourly", 30],
	["Brand X", "2000", "Modem", "Salaried", 30],
	["Brand Y", "2000", "Internal", "Contr.", 6],
	["Brand Y", "NT", "Internal", "Part-Time", 60],
	["Brand Y", "XP", "Modem", "Salaried", 15],
	["Brand X", "98", "Modem", "Part-Time", 60],
	["Brand X", "XP", "Modem", "Salaried", 60],
	["Brand X", "2000", "Internal", "Part-Time", 15],
	["Brand X", "2000", "Modem", "Contr.", 60],
	["Brand X", "98", "Modem", "Salaried", 10],
	["Brand X", "98", "Modem", "Part-Time", 30],
	["Brand X", "NT", "Modem", "Part-Time", 10],
	["Brand Y", "NT", "Modem", "Salaried", 60],
	["Brand Y", "NT", "Modem", "Hourly", 15],
	["Brand Y", "NT", "Modem", "Hourly", 30],
	["Brand Y", "NT", "Modem", "Hourly", 60],
	["Brand Y", "NT", "Modem", "Hourly", 10],
]
```

Correct me if I'm wrong, but if I'm getting it right, when we say `n = 3`, we want all triple-wise combinations to be tested. So in the result we want to see, for example, following `Brand X - NT - ...` combinations:
```
Brand X - NT - Internal
Brand X - NT - Modem [+]
Brand X - NT - Salaried 
Brand X - NT - Hourly
Brand X - NT - Part-Time [+]
Brand X - NT - Contr.
Brand X - NT - 6
Brand X - NT - 10 [+]
Brand X - NT - 15
Brand X - NT - 30
Brand X - NT - 60
...
```
But currently there are only 3 of them.

Filtering fix also fixed this issue,